ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_volume_collection.py

```diff
@@ -0,0 +1,722 @@
+    description:
+    - User defined key-value pairs that augment an volume collection attributes. List of key-value pairs. Keys must be unique and non-empty.
+    - When creating an object, values must be non-empty. When updating an object, an empty value causes the corresponding key to be removed.
+  name:
+    required: True
+    type: str
+    description:
+    - Name of volume collection.
+  no_reverse:
+    required: False
+    type: bool
+    default: False
+    description:
+    - Do not automatically reverse direction of replication.
+      Using this argument will prevent the new owner from automatically replicating the volume collection to this node when the handover completes.
+  override_upstream_down:
+    required: False
+    type: bool
+    description:
+    - Allow the handover request to proceed even if upstream array is down. The default behavior is to return an error when upstream is down.
+      This option is applicable for synchronous replication only.
+  promote:
+    required: False
+    type: bool
+    description:
+    - Take ownership of the specified volume collection. The volumes associated with the volume collection will be set to online and be
+      available for reading and writing. Replication will be disabled on the affected schedules and must be re-configured if desired. Snapshot
+      retention for the affected schedules will be set to the greater of the current local or replica retention values. This operation is not
+      supported for synchronous replication volume collections.
+  prot_template:
+    required: False
+    type: str
+    description:
+    - Name of the protection template whose attributes will be used to create this volume collection.
+      This attribute is only used for input when creating a volume collection and is not outputed.
+  replication_partner:
+    required: False
+    type: str
+    description:
+    - Name of the new volume collection owner.
+  replication_type:
+    choices:
+    - periodic_snapshot
+    - synchronous
+    required: False
+    type: str
+    default: periodic_snapshot
+    description:
+    - Type of replication configured for the volume collection.
+  state:
+    required: True
+    choices:
+    - present
+    - absent
+    - create
+    type: str
+    description:
+    - Choice for volume collection operations.
+  vcenter_hostname:
+    required: False
+    type: str
+    description:
+    - VMware vCenter hostname.
+  vcenter_username:
+    required: False
+    type: str
+    description:
+    - Application VMware vCenter username. String of up to 80 alphanumeric characters, beginning with a letter.
+      It can include ampersand (@), backslash (\), dash (-), period (.), and underscore (_).
+  vcenter_password:
+    required: False
+    type: str
+    description:
+    - Application VMware vCenter password. A password with few constraints.
+extends_documentation_fragment: hpe_nimble
+short_description: Manage HPE Nimble Storage volume collection.
```

Comment:
How do you add a schedule?  Will that be a separate module?

Reply:
are you talking of scheduling a protection group?  yes is the answer in that case. there is a module to be developed called hpe_nimble_protection

Reply:
I'm talking about adding a protection schedule to a volume collection.  That module should be named hpe_nimble_protection_schedule

Reply:
Yes George. We do have a module to develop called hpe_nimble_protection_schedule. 
This has a an option to take a volcoll name.